### PR TITLE
Fixes #13120 - Modal headers and bodys display properly.

### PR DESCRIPTION
**Pull request by zjhuntin**
This change is meant to fix many of the modals. Right now they do not properly display any text provided from an interpolated object. 

This is an example of what these looked like before the change:
before1

This is an example of after:
after1

I also noticed that pluralization was not working properly in the modals, so I fixed that as well.

Here is what they looked like before this:
before2

Here is what they look like now:
after2

### CODE REVIEW

**Comment by jlsherrill** [comment]
@waldenraines mind reviewing?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/bulk/views/bulk-actions.html [hunk 1]
```diff
@@ -24,9 +24,19 @@ <h2 class="fl" translate>Bulk Actions</h2>
...
 
-      <div bst-modal="removeProducts()">
-        <div data-block="modal-header" translate>Remove {{ productTable.getSelected().length }} Products?</div>
-        <div data-block="modal-body" translate>Are you sure you want to remove the {{ productTable.getSelected().length }} products(s) selected?</div>
+      <div bst-modal="removeProducts()" model="productTable">
+        <div data-block="modal-header" 
+             translate
+             translate-n="productTable.getSelected().length"
+             translate-plural="Remove {{ productTable.getSelected().length }} products?">
+             Remove {{ productTable.getSelected().length }} product?
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/bulk/views/bulk-actions.html]: Nitpick:  if there is only one product do we even need to show the total?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/bulk/views/bulk-actions.html [hunk 2]
```diff
@@ -24,9 +24,19 @@ <h2 class="fl" translate>Bulk Actions</h2>
...
+        </div>
+        <div data-block="modal-body" 
+             translate
+             translate-n="productTable.getSelected().length" 
+             translate-plural="Are you sure you want to remove {{ productTable.getSelected().length }} products selected?"> 
+             Are you sure you want to remove the {{ productTable.getSelected().length }} product selected?
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/bulk/views/bulk-actions.html]: Same nitpick here as above.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/views/product-repositories.html [hunk 3]
```diff
@@ -23,9 +23,19 @@
...
     <div data-block="actions">
-      <div bst-modal="removeSelectedRepositories()">
-        <div data-block="modal-header" translate>Remove {{ detailsTable.numSelected  }} Repositories?</div>
-        <div data-block="modal-body" translate>Are you sure you want to remove the {{ detailsTable.numSelected }} repositories selected?</div>
+      <div bst-modal="removeSelectedRepositories()" model="detailsTable">
+        <div data-block="modal-header" 
+             translate 
+             translate-n="detailsTable.numSelected" 
+             translate-plural="Remove {{ detailsTable.numSelected }} repostories?">
+          Remove {{ detailsTable.numSelected  }} repository?
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/views/product-repositories.html]: Same here, we should probably just say "Remove repository".

Comment (zjhuntin) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/views/product-repositories.html]: That's fair, I'll change all of these to not show the total when there is only one.

**Comment by zjhuntin** [comment]
@waldenraines Okay fixed to read better than it did before.

**Comment by waldenraines** [comment]
Looks good, thanks @zjhuntin!

**Pull request merged**